yamale/validators/constraints.py

```diff
@@ -115,3 +118,22 @@ def _is_valid(self, value):
 
     def _fail(self, value):
         return self.fail % (value, self._failed_char)
+
+
+class IpVersion(Constraint):
+    keywords = {'version': int}
+    fail = 'IP version of %s is not %s'
+
+    def _is_valid(self, value):
```

Comment:
I'm not particularly thrilled with this implementation of the `IpVersion` constraint.  It's almost the same as the `yamale.validators.validators.Ip.ip_address` method.  I implemented a version of it that deduplicated code by importing `yamale.validators.validators.Ip`, and that works fine for Python3 but not for Python2.  I guess Python3 handles recursive imports but Python2 doesn't?  Not sure.

I also thought about splitting the IP-specific bits into its own module, but I wasn't sure how you'd feel about that, so I just opted to leave things as-is and ask for some feedback.

Reply:
You could move both try/excepts to the utils file under a function `to_ipaddress` and just call it from both places. If calling ipaddress becomes a bottleneck you can always memoize that function so it's only ever called once per unique string. However, I think this is fine.

Reply:
I'll leave as-is, then.  Thanks!  :)